Automattic/jetpack
# Publicize: add Fediverse creator meta tag from Mastodon connection

Fixes #38565

## Proposed changes:

> [!NOTE]
> This is a follow-up to #38198, which was partially reverted in #38612.
> It's an alternative to #38565, to ensure better compatibility with WordPress.com Simple.

1. It takes into account the ActivityPub plugin, and ensure we're not adding a tag when the ActivityPub plugin already adds one.
2. It takes into account that on WordPress.com Simple, `get_services()` may return a `Keyring_Access_Token` object instead of an array.
3. It adds the 2 necessary methods directly to the `functions.opengraph.php` file, since we know it is loaded on the frontend on WordPress.com Simple (see conversation here to find out more: https://github.com/Automattic/jetpack/pull/38565#issuecomment-2277571635 )

### Other information:


## Jetpack product discussion

* peLCyI-nZ-p2

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> [!WARNING]
> Make sure you test this PR on WordPress.com Simple as wel as self-hosted.
> Also check your error logs to ensure that there are no notices or warnings.

* Go to Jetpack > Settings > Sharing
* Enable Jetpack Social, and connect your site to a test Mastodon account.
* Go to Posts > Add New
* Write a post and publish it. In the post sidebar, ensure the post is shared to Mastodon.
* Check the source of the post on the frontend: you should see this: `<meta name="fediverse:creator" content="@tests@fedi.jeremy.hu" />`
* Disable Jetpack Social
* Ensure the meta tag is gone, and that there are no errors on your site
* Enable Jetpack Social again.
* Now install and activate the ActivityPub plugin on your site.
    * > On WordPress.com Simple, you'll want to go to Tools > Marketing > Connections, and flip the toggle to "enter the fediverse".
* Check that same post again ; the meta tag added by Jetpack should be gone.
